openvox/install.md

```diff
@@ -50,28 +53,62 @@ Then install the packages you want.
-* If you're using distro provided packages, then you might have a bigger job.
-  For example, if you're using Debian packages, you may have several Puppet modules packaged as `.deb` packages that you'll have to move to your `Puppetfile`.
-  You might consider waiting until your distro packages OpenVox.
-    * If you do want to migrate now, then remove any puppet packages and dependencies you have installed using your distro tools.
-        * Debian family
-          * `apt autoremove <packagename>`
-        * RedHat family
-          * `yum autoremove <packagename>`
-    * You might also consider (carefully) cleaning up unused dependencies afterwards by running `apt` or `yum` autoremove without a package name.
+There are two ways to install Foreman: via the `foreman-installer` or via the `theforeman-foreman` module. ***In both cases, you must install OpenVox first***.
 
-You do not need to remove the `[apt|yum].puppet.com` repositories although the only thing you'll be able to use them for going forward is installing historical Puppet releases.
+#### Using the `foreman-installer`
+
+Edit `/etc/foreman-installer/custom-hiera.yaml` to include these Hiera settings.
+If you intend to continue managing your Foreman installation with puppet code, then follow the instructions in the [Managing OpenVox with OpenVox](#managing-openvox-with-openvox) sections to persist these settings into your standard Hiera data files.
+
+``` yaml
+---
+puppet::client_package: openvox-agent
+puppet::server_package: openvox-server
+puppetdb::puppetdb_package: openvoxdb
+puppetdb::master::config::terminus_package: openvoxdb-termini
+foreman::providers::oauth: false
+```
+Then manually install the `oauth` gem into OpenVox's gempath.
+
+```
+# /opt/puppetlabs/puppet/bin/gem install oauth
+```
+
+And finally, use [Foreman's Quickstart Guide](https://theforeman.org/manuals/3.15/quickstart_guide.html) to install Foreman.
+
+#### Using the `theforeman-foreman` module
+
+1. Ensure that OpenVox is installed, then follow the instructions in the [Managing OpenVox with OpenVox](#managing-openvox-with-openvox) sections to configure Hiera properly.
+1. Add `theforeman-foreman` to your `Puppetfile` and deploy it.
+1. Use the [documentation](https://github.com/theforeman/puppet-foreman) to classify your server node to your requirements and then run the OpenVox agent to configure everything.
+
+### Porting a Foreman infra to OpenVox
+
+If you're already running Foreman and you want to switch it to be backed by OpenVox then you have a little bit of housekeeping to do first.
+First, use your system package manager to remove the `puppet-agent-oauth` package.
+If you use the `foreman-installer` to manage and upgrade your setup, then ensure that you remove the cached scenario file, or it will continue to use old package names.
+
+```
+# rm /etc/foreman-installer/scenarios.d/last_scenario.yaml
+```
```

Comment:
This is never a good idea. It's a symlink pointing to the real scenario. This deselects the scenario, but that really isn't of any use. Especially if you only have a single scenario available.

Nor is it needed. The scenario file is describing the scenario to be used. The answers file (which you're not using) is 
```suggestion
```

Reply:
I'm not convinced this it's enough to specify the CLI parameters. I can validate later, but what happens now is that someone on Debian does

1. apt install foreman-installer (which pulls in Debian packaged `puppet-agent`, non-AIO)
2. configures hiera data (which I believe is roughly equivalent to passing CLI args)
3. runs the installer
4. everything breaks because `/etc/puppet/<stuff>` exists even after the `openvox-agent` package replaces the system `puppet-agent` package and foreman-installer continues to use the `/etc/puppet` directory.

Even after deleting the `/etc/puppet` directory it still breaks, even with the proper package names. That's what we're trying to address here.

Reply:
Yes, if you install the proper package to begin with, then it works great. But we can't require perfection from our users! The only way to get from system `puppet-agent` -> AIO `openvox-agent` is to start with a fresh answers file or to edit some 500 lines of data and correct each and every legacy path.